lib/checkleakautovar.cpp

```diff
@@ -671,9 +671,10 @@ bool CheckLeakAutoVar::checkScope(const Token * const startToken,
                 tok = tok->tokAt(3);
             else
                 tok = tok->next();
-            if (tok->str() == "(")
+            bool startparen;
+            if ((startparen = (tok->str() == "(")))
                 tok = tok->next();
-            while (Token::Match(tok, "%name% ::|."))
+            while (Token::Match(tok, "%name% ::|.") || (startparen && Token::Match(tok, "%name% ,")))
```

Comment:
it's really unfortunate that we didn't use the AST here. but well.. unless you would rewrite this , this looks like a reasonable solution.